service/gcs/errors/errors.go

```diff
@@ -45,3 +62,111 @@ func (e *processDoesNotExistError) Error() string {
 func NewProcessDoesNotExistError(pid int) *processDoesNotExistError {
 	return &processDoesNotExistError{Pid: pid}
 }
+
+// StackTracer is an interface originating (but not exported) from the
+// github.com/pkg/errors package. It defines something which can return a stack
+// trace.
+type StackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+type baseHresultError struct {
+	hresult Hresult
+}
+
+func (e *baseHresultError) Error() string {
+	return fmt.Sprintf("HRESULT: %d", e.Hresult())
+}
+func (e *baseHresultError) Hresult() Hresult {
+	return e.hresult
+}
+func (e *baseHresultError) Format(s fmt.State, verb rune) {
```

Comment:
Yeah, it might be. I wanted to be explicit, since I defined Format on wrappingHresultError here. For the default formatting behavior to call the Error() method on this, would I need to define a String() method? I could replace this Format method with a String method if that seems correct.